tests/lazy/test_sam_knn.py

```diff
@@ -73,7 +77,7 @@ def test_sam_knn_coverage(package_path):
     wait_samples = 20
 
     while cnt < max_samples:
-        X, y = stream.next_sample()
+        X, y = X_gen.next_sample(), y_gen.next_sample()
```

Comment:
Why was it necessary to split into two what was previously a single data source?